azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -1025,8 +1025,11 @@ public String submitExecutableFlow(ExecutableFlow exflow, String userId)
           try {
             dispatch(reference, exflow, choosenExecutor);
           } catch (ExecutorManagerException e) {
-            executorLoader.removeActiveExecutableReference(reference
-              .getExecId());
+            // When flow dispatch fails, should update the flow status
+            // to FAILED in execution_flows DB table as well. Currently
+            // this logic is only implemented in multiExecutorMode but
```

Comment:
Yes, currently the inconsistency issue already exists between active_executing_flows and executionf_flows table. After deprecating active_executing_flows table, this issue will be exposed in single executor mode when dispatch fails. So it is necessary to set the flow status to FAILED in execution_flows table when dispatch fails.